src/main/frontend/handler/paste.cljs

```diff
@@ -72,6 +72,19 @@
                                              (gp-util/safe-decode-uri-component text))]
     (try-parse-as-json (second matched-text))))
 
+(defn- selection-within-link?
+  [selection-and-format]
```

Comment:
Thanks for the great fix!
Can we add some unit tests to avoid potential broken in future?
For how unit test works in the Logseq project: https://github.com/logseq/logseq/blob/master/docs/dev-practices.md#unit-testing

Reply:
Done, and add "If the selected content and content from the clipboard are both URL, replace it"